main.go

```diff
@@ -175,24 +190,18 @@ func main() {
 			log.Printf("Dumped system information to: %s", basePath)
 			return
 		}
-		cmd = exec.Command("rm", "-rf", basePath)
-		cmd.Run()
+
+		// The archive was created successfully, remove basePath. The
+		// error from os.RemoveAll is intentionally ignored, since there
+		// is no useful action we can do, and we don't need to confuse
+		// the user with an error message.
+		os.RemoveAll(basePath)
 
 		log.Printf("Dumped system information to: %s", basePath+".tar.gz")
 	}()
 
 	log.Println("Running tasks...")
 
-	// Avoid the creating goroutines and other controls if we're executing
-	// tasks sequentially.
-	if *maxParallelTasks == 1 {
```

Comment:
Yes, the special case was there as a "safe way to run the tool without the concurrency stuff", but instead of doing good it was just accumulating bugs.
The number of parallel tasks defaults to the number of CPUs, so I've mostly been testing the tool with >1 values.